internal/context/prompts.go

```diff
@@ -31,6 +44,14 @@ func KeyMatchCondition(key string, value string) func(map[string]string) bool {
 	}
 }
 
+type CustomConditionSignature func(map[string]string) bool
+
+func CustomCondition(fn CustomConditionSignature) func(map[string]string) bool {
```

Comment:
You should be able to specify the return type as `CustomConditionSignature` as well.